opencensus/proto/agent/common/v1/common.proto

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+// NOTE: This proto is experimental and is subject to change at this point.
+// Please do not use it at the moment.
+
+package opencensus.proto.agent.common.v1;
+
+import "google/protobuf/timestamp.proto";
+
+option java_multiple_files = true;
+option java_package = "io.opencensus.proto.agent.common.v1";
+option java_outer_classname = "CommonProto";
+
+option go_package = "github.com/census-instrumentation/opencensus-proto/gen-go/agentproto/commonproto/v1";
+
+message Node {
+  // Identifier that uniquely identifies a process within a VM/container.
+  ProcessIdentifier identifier = 1;
+
+  // Information on the OpenCensus Library who initiates the stream.
+  LibraryInfo LibraryInfo = 2;
+
+  // Additional attributes.
+  map<string, string> attributes = 3;
+
+  // Additional informantion on service.
+  ServiceInfo serviceInfo = 4;
+
+  // TODO(songya): Add more identifiers in the future as needed, like cloud identifiers.
+}
+
+// Identifier that uniquely identifies a process within a VM/container.
+// In the future we plan to extend the identifier proto definition to support
+// additional information (e.g cloud id, monitored resource, service name, etc.)
+message ProcessIdentifier {
+
+  // The host name. Usually refers to the machine/container name.
+  // For example: os.Hostname() in Go, socket.gethostname() in Python.
+  string host_name = 1;
+
+  // Process id.
+  uint32 pid = 2;
+
+  // Represented in epoch time.
+  google.protobuf.Timestamp timestamp = 3;
+
+  // TODO(songya): Add more fields in the future as needed.
+}
+
+// Information on OpenCensus Library.
+message LibraryInfo {
+
+  enum Language {
+    UNKNOWN = 0;
+    GO_LANG = 1;
+    JAVA = 2;
+    PYTHON = 3;
+    RUBY = 4;
+    PHP = 5;
+    C_SHARP = 6;
+    CPP = 7;
+    ERLANG = 8;
+    NODE_JS = 9;
+  }
+
+  // Language of OpenCensus Library.
+  Language language = 1;
+
+  // Version of Agent exporter of Library.
+  string version = 2;
+}
+
+// Additional service information.
+message ServiceInfo {
+
+  // Name of the service.
+  string name = 1;
+
+  // Version of the service.
+  string version = 2;
```

Comment:
Version was just an example. Please don't add it for the moment.

Reply:
Done.